numpy/f2py/__init__.py

```diff
@@ -59,10 +64,29 @@ def compile(source,
         f.write(source)
         f.flush()
 
-        args = ' -c -m {} {} {}'.format(modulename, f.name, extra_args)
-        c = '{} -c "import numpy.f2py as f2py2e;f2py2e.main()" {}'
-        c = c.format(sys.executable, args)
-        status, output = exec_command(c)
+        args = ['-c', '-m', modulename, f.name]
+
+        is_posix = (os.name == 'posix')
```

Comment:
Condensing this to 1 line as suggest is probably fine -- just keep in mind that the 100 % coverage is slightly optimistic since we are no longer being punished for not exploring the Windows code path with coverage active; as long as appveyor is green it should likely be safe though.

Reply:
That's an interesting point that I hadn't considered - but this isn't being covered anyway because `extra_args` is never used, right?

Reply:
Oh, also - this can move inside the `if` below